File: src/main/java/pr/eleks/we_at_her/services/data/UserService.java
package pr.eleks.we_at_her.services.data;

import pr.eleks.we_at_her.dto.UserDto;
import pr.eleks.we_at_her.entities.User;

public interface UserService {
    UserDto createUser(UserDto userDto);

    UserDto convertToDto(User user);

    User convertToEntity(UserDto userDto);

    UserDto findUserByUuid(String uuidString);

    UserDto activateUserByUuid(String uuidString);
}

File: src/main/kotlin/pr/eleks/we_at_her/services/data/UserService.kt
package pr.eleks.we_at_her.services.data

import org.springframework.security.core.userdetails.UserDetailsService
import pr.eleks.we_at_her.dto.UserDto
import pr.eleks.we_at_her.entities.User
import pr.eleks.we_at_her.exceptions.UUIDNotFoundException

interface UserService : UserDetailsService {
    fun createUser(userDto: UserDto): UserDto
    fun activateUserByUuid(uuidString: String): UserDto
    fun findUserByUuid(uuidString: String): UserDto?
    fun convertToDto(user: User): UserDto
    fun convertToEntity(userDto: UserDto): User
}